src/mango_idx_text.erl

```diff
@@ -327,3 +328,81 @@ indexable_fields(Fields, {op_null, {_, _}}) ->
 
 indexable_fields(Fields, {op_default, _}) ->
     [<<"$default">> | Fields].
+
+
+maybe_reject_index_all_req({Def}, #db{name=DbName, user_ctx=Ctx}) ->
+    User = Ctx#user_ctx.name,
+    Fields = couch_util:get_value(fields, Def),
+    case {Fields, forbid_new_index_all()} of
+        {all_fields, "true"} ->
+            ?MANGO_ERROR(forbid_new_index_all);
+        {all_fields, "warn"} ->
+            couch_log:warning("User ~p is indexing all fields in db ~p",
+                [User, DbName]);
+        _ ->
+            ok
+    end.
+
+
+forbid_new_index_all() ->
+    config:get("mango", "forbid_new_index_all", "false").
```

Comment:
strike 'new', it only sounds like it qualifies because this feature doesn't exist yet.